litellm/llms/azure.py

```diff
@@ -989,24 +989,55 @@ def embedding(
                     status_code=422, message="max retries must be an int"
                 )
 
-            # init AzureOpenAI Client
-            azure_client_params = {
-                "api_version": api_version,
-                "azure_endpoint": api_base,
-                "azure_deployment": model,
-                "http_client": litellm.client_session,
-                "max_retries": max_retries,
-                "timeout": timeout,
-            }
-            azure_client_params = select_azure_base_url_or_endpoint(
-                azure_client_params=azure_client_params
-            )
-            if api_key is not None:
-                azure_client_params["api_key"] = api_key
-            elif azure_ad_token is not None:
-                if azure_ad_token.startswith("oidc/"):
-                    azure_ad_token = get_azure_ad_token_from_oidc(azure_ad_token)
-                azure_client_params["azure_ad_token"] = azure_ad_token
+            # Check if it's Cloudflare AI Gateway
+            if "gateway.ai.cloudflare.com" in api_base:
```

Comment:
hey @Manouchehri can we move away from hardcoded api bases - e.g. gateway.ai.cloudflare.com? 

a more generic implementation where api_base is set and used, would help simplify the code here

Reply:
i understand, but can we move this to just a simple check if api_base is set, if so -> set the base_url

if not, don't?